Try smaller up/down arrows, and focus management

jasmussen opened the pull request:
This PR proposes to reduce the visual and cognitive weight of the up/down movers. It is part of an ongoing effort to improve the writing flow by refining the UI. Additional efforts include a recent visual change to the hover/active/focus states of buttons, and in #4539 those efforts include unifying where the contextual inserter sits and how it behaves.

Screenshots:

<img width="299" alt="hover" src="https://user-images.githubusercontent.com/1204802/35217709-86a42dca-ff6c-11e7-8dd1-48cca346be36.png">

<img width="217" alt="active" src="https://user-images.githubusercontent.com/1204802/35217716-89d6b238-ff6c-11e7-986f-363d9b329c2b.png">

<img width="206" alt="keyboard" src="https://user-images.githubusercontent.com/1204802/35217719-8cbcbf92-ff6c-11e7-9781-5f13dad45b7d.png">

GIF:

![in-action](https://user-images.githubusercontent.com/1204802/35217721-8f701c34-ff6c-11e7-99c1-5f4654d2894a.gif)

It reduces the visual weight by making the icons smaller (note that the hit-area size is unchanged), and by handling the focusring so it's there for keyboard interactions but not for mouse interactions (<a href="https://www.youtube.com/watch?v=ilj2P5-5CjI">see also</a>).

Note that:

- the focusring is present for keyboard users
- the focusring change should not apply to other buttons in the UI, only the up/down arrows
- there will be direct keyboard shortcuts for moving blocks up/down (see #71)
- drag & drop is in development to further expand the ways blocks can be rearranged

## Why this change?

Gutenberg has received a lot of fair feedback around the UI feeling distracting and impacting the writing flow. While we have done many things to address this, including fading out UI when writing, simplifying toolbars and hover styles, we are approaching the phase where we refine the subtle details. In this case, I'm proposing that the value the up/down buttons add to the writing and editing flow is well-balanced with the smaller icons and lack of mousefocus ring.

## Tests

This has been tested for mouse and keyboard interactions in:

- Mac/Chrome
- Mac/Safari
- Mac/Firefox
- Windows/Chrome
- Windows/Edge
- Windows/IE11

Penny for your thoughts.

Commit:
Try new icons for movers

These icons are smaller and less obtrusive, which makes them less heavy to look at.

This PR also employs a workaround to achieve the effect that `:focusring` would, were it implemented (see https://www.youtube.com/watch?v=ilj2P5-5CjI). This is applied only to the mover icons.

GIF: https://cldup.com/5lyhf0iSTp.gif

karmatosed commented:
Thank you for doing this change. I think it's incredibly important for us to look at the UI and balance with the accessibility requirements. I feel based on other examples, this is a solid balance. I know you're really researched a lot here and props to you for that @jasmussen.

Commit:
Adjust test to not look for the icon

The icon is now an SVG passed on

jasmussen commented:
I fixed the tests. #4539 depends on this getting merged so it can be rebased and fixed. It's pretty vital for the writing flow we get this in as soon as possible. As such, given the approval, I'm going to merge this now and continue work. If there are objections to the changes, it's always possible to look at those separately.
